apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditLimitPrice.tsx

```diff
@@ -1,69 +1,78 @@
 /* eslint-disable react/jsx-newline */
 import {useSwap} from '@yoroi/swap'
 import BigNumber from 'bignumber.js'
-import React from 'react'
+import React, {useState} from 'react'
 import {StyleSheet, Text, TextInput, View} from 'react-native'
 
-import {Spacer} from '../../../../../components'
 import {useLanguage} from '../../../../../i18n'
 import {useSelectedWallet} from '../../../../../SelectedWallet'
 import {COLORS} from '../../../../../theme'
 import {useTokenInfo} from '../../../../../yoroi-wallets/hooks'
 import {Quantities} from '../../../../../yoroi-wallets/utils'
+import {getBuyQuantityForLimitOrder} from '../../../common/helpers'
 import {useStrings} from '../../../common/strings'
 import {useSwapTouched} from './TouchedContext'
+const BORDER_SIZE = 1
+const PRECISION = 10
 
 export const EditLimitPrice = () => {
   const strings = useStrings()
   const {numberLocale} = useLanguage()
 
   const wallet = useSelectedWallet()
 
-  const {createOrder} = useSwap()
-  // TODO: limitChanged should be defined in useSwap
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const limitChanged = (_todo: any) => null
+  const {createOrder, limitPriceChanged, buyAmountChanged} = useSwap()
+  const sellTokenInfo = useTokenInfo({wallet, tokenId: createOrder.amounts.sell.tokenId})
+  const buyTokenInfo = useTokenInfo({wallet, tokenId: createOrder.amounts.buy.tokenId})
 
   const {isBuyTouched, isSellTouched} = useSwapTouched()
 
-  const price =
+  const defaultPrice =
     isBuyTouched &&
     isSellTouched &&
     createOrder.selectedPool?.price !== undefined &&
     !Number.isNaN(createOrder.selectedPool.price)
       ? createOrder.selectedPool.price
       : 0
 
-  const formattedPrice = new BigNumber(price).decimalPlaces(6).toString(10)
-
-  const [inputValue, setInputValue] = React.useState(formattedPrice)
+  const limitPrice = createOrder.limitPrice !== undefined ? BigNumber(createOrder.limitPrice).toNumber() : defaultPrice
 
   const tokenToSellInfo = useTokenInfo({wallet, tokenId: createOrder.amounts.sell.tokenId})
-  const tokenToSellName = isSellTouched ? tokenToSellInfo.ticker ?? tokenToSellInfo.name : '-'
   const tokenToBuyInfo = useTokenInfo({wallet, tokenId: createOrder.amounts.buy.tokenId})
+
+  const tokenToSellName = isSellTouched ? tokenToSellInfo.ticker ?? tokenToSellInfo.name : '-'
   const tokenToBuyName = isBuyTouched ? tokenToBuyInfo.ticker ?? tokenToBuyInfo.name : '-'
+  const formattedValue = BigNumber(limitPrice).toFormat(numberLocale)
+
+  const [text, setText] = useState(formattedValue)
```

Comment:
what is text? is it the inputValue?
```suggestion
  const [text, setText] = React.useState(formattedValue)
```